### PR TITLE
Extract inmemory from core and backend testutils form test

### LINKED ISSUE #1863: Move inmemory backend out of janusgraph-core
**Describe the feature:**
All other backends are an extra module beside inmemory. Now, we have a PR to make the inmemory backend production-ready. It would good to be in an extra module.

**Describe a specific use case for the feature:**
All backends should be in an extra module.

### PR DESCRIPTION
**Pull request by farodin91**
Fixes #1863

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you written and/or updated unit tests to verify your changes?
- [x] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [x] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [x] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -85,8 +77,8 @@ jobs:
       env: MODULE='berkeleyje'
     - <<: *STANDARD_TEST_JOB
       env: MODULE='test'
-    - <<: *TINKERPOP_TEST_JOB
```
Comment (porunov) [inline .travis.yml]: Why tinkerpop test job is removed?

Comment (farodin91) [reply .travis.yml]: I included tinkerpop test execution directly into the inmemory tests. 

If we add tinkerpop test execution for other backends, we can add this again.

**Comment by dk-github** [comment]
LGTM.

Is there a way to run a verification build (i.e. this change on top of master) and assert it is green before actually merging?

**Comment by farodin91** [comment]
@dk-github I rebased the commit on top of the master.

**Comment by dk-github** [comment]
As I understood, formally speaking this can be merged right now since it has 1 approval and your approval is implicit?..

I propose to wait a couple more days, and if no further comments, merge it on Monday next week.

Then I will try to carve out the time to rebase #1483 on top and fix the review comments...

**Pull request merged**